tests/ui/consts/issue-102117.rs

```diff
@@ -17,10 +17,9 @@ impl VTable {
             &VTable {
                 layout: Layout::new::<T>(),
                 type_id: TypeId::of::<T>(),
-                //~^ ERROR the parameter type `T` may not live long enough
-                //~| ERROR the parameter type `T` may not live long enough
                 drop_in_place: unsafe {
                     transmute::<unsafe fn(*mut T), unsafe fn(*mut ())>(drop_in_place::<T>)
+                    //~^ ERROR can't drop `T` in const contexts
```

Comment:
Isn't this a bad error? We're not actually *calling* the function that needs const drop, and this breaks stable code:

```rust
pub const fn get_drop<T>() -> unsafe fn(*mut T) {
    core::ptr::drop_in_place::<T>
}
```

Reply:
Similarly for `array::from_fn`:

```rust
pub const fn array_from_fn() -> fn(fn(usize) -> ()) -> [(); 42] {
    core::array::from_fn
}
```

... works on stable, but fails here for lack of `~const FnMut` on the function pointer.

Reply:
Does this go back to the previous error after #109557?

Reply:
The const-eval working group has been discussing the status of const traits in the compiler as well as the standard library. We came to a consensus that all uses of `~const` in the standard library should be temporary removed to allow for a clean impl of it to land in the compiler. See [this zulip thread](https://rust-lang.zulipchat.com/#narrow/stream/146212-t-compiler.2Fconst-eval/topic/.60const.20Trait.60.20removal.20or.20rework).